init-interfaces.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+
+set -ex
+
+if [[ ! -f /boot/mac_addresses ]] ; then
+  echo "no mac address configuration file found .. exiting"
+  exit 1
+fi
+
+primary_mac=$(cat /boot/mac_addresses | awk -F= '/PRIMARY_MAC/ {print $2}')
+secondary_mac=$(cat /boot/mac_addresses | awk -F= '/SECONDARY_MAC/ {print $2}')
+
+default_device=""
+secondary_device=""
+default_connection_name=""
+secondary_connection_name=""
+
+for dev in $(nmcli device status | awk '/ethernet/ {print $1}'); do
+  dev_mac=$(nmcli -g GENERAL.HWADDR dev show $dev | sed -e 's/\\//g' | tr '[A-Z]' '[a-z]')
+  case $dev_mac in
+    $primary_mac)
+      default_device=$dev
+      default_connection_name=$(nmcli -g GENERAL.CONNECTION dev show $dev)
+      ;;
+    $secondary_mac)
+      secondary_device=$dev
+      secondary_connection_name=$(nmcli -g GENERAL.CONNECTION dev show $dev)
+      ;;
+    *)
+      ;;
+   esac
+done
+
+echo -e "default dev: $default_device ($default_connection_name)\nsecondary dev: $secondary_device ($secondary_connection_name)"
+
```

Comment:
Setting it up is not enough, let's be explicit about the IPv4 method being auto and DHCP ID being MAC: https://github.com/RHsyseng/rhcos-slb/blob/main/setup-ovs.sh#L49

Reply:
I've changed the script to also configure if not already configured